nbgrader/preprocessors/clearmarkingscheme.py

```diff
@@ -50,6 +60,7 @@ def _remove_mark_scheme_region(self, cell: NotebookNode) -> bool:
         new_lines = []
         in_ms = False
         removed_ms = False
+        attachment_regex = r"!\[(?:.*?)\]\((?:attachment:.+?)\)"
```

Comment:
You are right that the regular expression can be simpler. Don't know what I was thinking with the groups.

I think the "?"s following "*"s should still be there though to match properly. Otherwise, if there are two attachments in a cell, everything in between them will be in the match as well. Though functionally, that doesn't change anything right now, since this code aims only to detect their presence and warn.

Reply:
OK got it, thank you for the pointer.
I was providing a direct link to the image like `[...](./image.png)`, which makes no sense for this issue, as the image would be released anyway if it is in the same directory. I didn't know about the mechanism of drag&drop the image.